src/NuGet.Core/NuGet.Packaging/Signing/Utility/CertificateUtility.cs

```diff
@@ -239,19 +240,68 @@ public static byte[] GetHash(X509Certificate2 certificate, HashAlgorithmName has
         }
 
         /// <summary>
-        /// Determines if a certificate is self-signed.
+        /// Determines if a certificate is self-issued.
         /// </summary>
-        /// <param name="certificate">The certificate to check.</param>
-        /// <returns>A boolean flag indicating if the certificate is self-signed.</returns>
+        /// <remarks>Warning:  this method does not evaluate certificate trust, revocation status, or validity!</remarks>
+        /// <param name="certificate"></param>
+        /// <returns><c>true</c> if the certificate is self-issued; otherwise, <c>false</c>.</returns>
         /// <exception cref="ArgumentNullException">Thrown if <paramref name="certificate" /> is <c>null</c>.</exception>
-        public static bool IsSelfSigned(X509Certificate2 certificate)
+        public static bool IsSelfIssued(X509Certificate2 certificate)
         {
             if (certificate == null)
             {
                 throw new ArgumentNullException(nameof(certificate));
             }
 
-            return certificate.IssuerName.RawData.SequenceEqual(certificate.SubjectName.RawData);
+            using (var chainHolder = new X509ChainHolder())
```

Comment:
👍 nice! Is this doing any online check when building the chain for non-self signed certificates?